pkg/tfbridge/detailed_diff.go

```diff
@@ -259,6 +292,78 @@ func (differ detailedDiffer) makeListDiff(
 	return diff
 }
 
+type setChangeIndex struct {
+	engineIndex   int
+	newStateIndex int
+	oldChanged    bool
+	newChanged    bool
+}
```

Comment:
If this type is only used in `makeSetDiff`, can we move it's definition into `makeSetDiff`?